std/experimental/testing/gen_ut_main.d

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/rdmd
+
+/**
+ * Implements a program to search a list of directories
+ * for all .d files, then writes a D program to run all
+ * unit tests in those files using std.experimental.testing.
+ *
+ * The resulting program has a few command-line arguments,
+ * the most important of which is -f to select the name
+ * of the output file. Use -h to obtain help.
+ */
+
+import std.stdio;
+import std.array: replace, array, join;
+import std.conv: to;
+import std.algorithm: map;
+
+
+int main(string[] args)
+{
+    const options = getOptions(args);
+    if (options.help || options.showVersion) return 0;
+
+    writeFile(options, findModuleNames(options.dirs));
+
+    return 0;
+}
+
+private struct Options
+{
+    bool verbose;
+    string fileName;
+    string[] dirs;
+    bool help;
+    bool showVersion;
+}
+
+private Options getOptions(string[] args)
+{
+    import std.getopt;
+
+    Options options;
+    auto getOptRes = getopt(
+        args,
+        "verbose|v", "Verbose mode.", &options.verbose,
+        "file|f", "The filename to write. Will use a temporary if not set.", &options.fileName,
+        "version", "Show version.", &options.showVersion,
+        );
+
+    if (getOptRes.helpWanted)
+    {
+        defaultGetoptPrinter("Usage: gen_ut_main [options] [test1] [test2]...",
+                             getOptRes.options);
+        options.help = true;
+        return options;
+    }
+
+    if (options.showVersion)
+    {
+        writeln("gen_ut_main version v0.2.5");
+        return options;
+    }
+
+    if (options.fileName)
+    {
+        import std.file: exists, remove;
+        if (exists(options.fileName)) remove(options.fileName);
+    }
+    else
+    {
+        options.fileName = createFileName(); //random filename
```

Comment:
I'm not sure if a random filename for the program is the best choice. 

why not something like __default_unittest_main.d

dmd -main -C creates a __main.o so you would be close and you can remove 17 lines